actions/telemetry_actions.jsx

```diff
@@ -92,22 +99,70 @@ export function measure(name1, name2) {
     return [lastDuration, measurementName];
 }
 
-export function trackLoadTime() {
+function trackPageLoadTimeEvent() {
     if (!isSupported([performance.timing.loadEventEnd, performance.timing.navigationStart])) {
         return;
     }
 
+    const {loadEventEnd, navigationStart} = window.performance.timing;
+    const pageLoadTime = loadEventEnd - navigationStart;
+    trackEvent('performance', 'page_load', {duration: pageLoadTime});
+}
+
+/**
+ * Measures the time and number of requests on first page load.
+ */
+export function measurePageLoadTelemetry() {
     // Must be wrapped in setTimeout because loadEventEnd property is 0
     // until onload is complete, also time added because analytics
     // code isn't loaded until a subsequent window event has fired.
     const tenSeconds = 10000;
     setTimeout(() => {
-        const {loadEventEnd, navigationStart} = window.performance.timing;
-        const pageLoadTime = loadEventEnd - navigationStart;
-        trackEvent('performance', 'page_load', {duration: pageLoadTime});
+        trackPageLoadTimeEvent();
+        trackNumOfRequestEvent('page_load');
     }, tenSeconds);
 }
 
+/**
+ * Clears all performance entries of entry type "resource".
+ */
+export function clearResourceTimings() {
+    if (!shouldTrackPerformance() || !SUPPORTS_RESOURCE_TIMINGS) {
+        return;
+    }
+
+    performance.clearResourceTimings();
+}
+
+/**
+ * Send an telemetry for number of requests to the server for an event.
+ * @param {String} initiatedBy The name of the event that initiated the tracking
+ */
+export function trackNumOfRequestEvent(initiatedBy) {
+    if (!shouldTrackPerformance() || !SUPPORTS_RESOURCE_TIMINGS) {
+        return;
+    }
+
+    let totalRequests = 0;
+    let maxResourceSize = 0; // in Bytes
+
+    performance.getEntriesByType('resource').forEach((entry) => {
+        if (entry.initiatorType === 'xmlhttprequest' || entry.initiatorType === 'fetch') {
+            totalRequests += 1;
+            maxResourceSize = Math.max(maxResourceSize, entry.encodedBodySize);
```

Comment:
does sending max resource size makes sense, i thought it would give an idea on how large resource is, ?

Reply:
Any other telemetry derived from resource timing api is can send ?

Reply:
I wonder if `encodedBodySize` or `decodedBodySize` would be more helpful to have here? I assume `encodedBodySize` includes like gzip compression if that's enabled on the server. I'm trying to think about which we'll care more about since encoded means that it'll be more realistic, but it'll then be influenced by server settings.

Then again, all of this will be influenced by server settings like CRT.